### PR TITLE
Handle deleted shards

### PR DESCRIPTION
**Pull request by itsvikramagr**
Addresses #55 

After re-sharding and end of the retention period, closed shards transitions to an EXPIRED state. Calls to getStreamDescription().getShards to enumerate the shards in the stream do not include EXPIRED shards in the list shards returned.
https://docs.aws.amazon.com/streams/latest/dev/kinesis-using-sdk-java-after-resharding.html

While constructing a new batch, we should exclude such expired shards. Added a source option "failOnDataLoss" which can be set to false if we don't want the application to fail.

### CODE REVIEW

**Comment by PranshuAdhikari** [comment]
@itsvikramagr : This fix doesn't exclude shards which have been removed from kinesis after closing. Getting an Exception "ResourceNotFoundException".

**Comment by itsvikramagr** [comment]
@PranshuAdhikari  -  Can you set "kinesis.failOnDataLoss" to false and try?

I will see how we can have a consistent way to specify the configs.

**Comment by PranshuAdhikari** [comment]
@itsvikramagr - I have tried with "failOnDataLoss" to false, getting the same exception "ResourceNotFoundException".

**Comment by itsvikramagr** [comment]
Can you try with `kinesis.failOnDataLoss` as false

**Comment by PranshuAdhikari** [comment]
yes i have tried with this aswell , getting the same exception.

**Comment by itsvikramagr** [comment]
can you share logs and offset info at my email-id. I will look at the issue and get back.

File: src/main/scala/org/apache/spark/sql/kinesis/ShardSyncer.scala [hunk 1]
```diff
@@ -181,21 +177,70 @@ private[kinesis] object ShardSyncer extends Logging {
...
+
   def getLatestShardInfo(
       latestShards: Seq[Shard],
       prevShardsInfo: Seq[ShardInfo],
-      initialPosition: KinesisPosition): Seq[ShardInfo] = {
+      initialPosition: KinesisPosition,
+      failOnDataLoss: Boolean = true): Seq[ShardInfo] = {
 
     if (latestShards.isEmpty) {
       return prevShardsInfo
```
Comment (Vinaypandit) [inline src/main/scala/org/apache/spark/sql/kinesis/ShardSyncer.scala]: Do we need to filter the previous shard list for deleted shards?

**Comment by PranshuAdhikari** [comment]
LGTM, working as expected.

**Pull request merged**